dali/python/nvidia/dali/auto_aug/core/_args.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+
+
+class MissingArgException(Exception):
+
+    def __init__(self, message, augmentation, missing_args):
+        super().__init__(message)
+        self.augmentation = augmentation
+        self.missing_args = missing_args
+
+
+class UnusedArgException(Exception):
+
+    def __init__(self, message, unused_args):
+        super().__init__(message)
+        self.unused_args = unused_args
+
+
+def filter_extra_accepted_kwargs(fun, kwargs, skip_positional=0):
```

Comment:
I am asking for docs as the name can be understood as removing the extra kwargs that are accepted rather than obtaining them.